### PR TITLE
fix(Windows): taking input from stdin on startup

### LINKED ISSUE #21776: Neovim on Windows hangs after reading from stdin
### Describe the bug

Starting neovim with text read from stdin through redirection causes neovim to hang.  Needs `Ctrl-C` to exit to the shell.

### Steps to reproduce

```
echo 'foo' | nvim
```

### Expected behavior

Neovim doesn't hang.

### Neovim version (nvim -v)

NVIM v0.9.0-dev-664+ga5f4ba74472f

### Vim (not Nvim) behaves the same?

no, vim 9.0.815

### Operating system/version

Windows 11 22H2

### Terminal name/version

Windows Terminal, Conhost

### $TERM environment variable

nil, xterm-256color

### Installation

Built from source

### PR DESCRIPTION
**Pull request by 3N4N**
Fixes #21776.

### CODE REVIEW

**Comment by bfredl** [comment]
This is deleting the code for linux. fixing the issue would mean adding an `#ifdef MSWIN` and add the missing code for windows.

**Comment by 3N4N** [comment]
Uh, understood.  I honestly thought they are unnecessary code; that's why I scrapped it.  But if re-assigning stdin to stderr is necessary (if that's what those two lines are doing, I'm not sure), then this PR should be closed.  I don't know if something similar is available in Windows's C libraries.

**Comment by 3N4N** [comment]
> I don't know if something similar is available in Windows's C libraries.

The same is applicable to Windows system as well.  File descriptors are the same. I don't know what's causing the problem then.

**Comment by bfredl** [comment]
Code is spread out between ui_client.c and tui/input.c and thus a bit hard to follow . I'll clean it up.
